### ISSUE TITLE
Remove DetailAST.branchContains

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#5234]
Split from https://github.com/checkstyle/checkstyle/issues/5124 ,

We should remove `branchContains` completely from `DetailAST`.

This issue is on hold until the other issue is finished and when we decide when we wish to break compatibility.

**Comment by romani** [comment]
We need to deprecate it, completely forbid to use it in our code by some Check and .... - all in scope of #5124

remove in several versions completely.

**Comment by rnveach** [comment]
@romani What do you think of also removing the Javadoc equivalent which also isn't used in checkstyle anywhere.
https://github.com/checkstyle/checkstyle/blob/6bab0788365ba9afb713e14d9f6263f1f60b7aa2/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java#L230

**Comment by rnveach** [comment]
This is blocked by https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/669 .

**Comment by romani** [comment]
From https://github.com/checkstyle/checkstyle/pull/7424#issuecomment-569995176

Can we have some/any utility method that can be used instead ? to mention it in migration notes and let users use it to keep functionality the same, even it is not efficient approach.

Update of legacy Checks might be very painful process, users will likely to agree on anything to keep it as is.

We will test such migration on sevntu Checks. If we succeed sevntu migration, we can sure that user will have barrable size of problems during upgrade.

**Comment by rnveach** [comment]
Yes we can always create a utility that behaves the same but then we will have to keep it in main repo and require it to meet all our requirements, 100% coverage and pitest passage. All this will do is kick the can down the road as users will just switch to the new utility and not rework this code to use this type of method. My aim was to completely remove this method.

If we are to build a duplicate of this then why not just leave the method as it is.

**Comment by romani** [comment]
>require it to meet all our requirements, 100% coverage and pitest passage.

Deprecation side effect stuff can have pure UTs, so coverage will be technical.

>If we are to build a duplicate of this then why not just leave the method as it is.

We remove it from API, deprecation in Util will be less accurate. We can remove it later on to force users just copy method code to their codebase if they want no heavy refactoring.

**Comment by nmancus1** [comment]
Can we just create a separate issue to deprecate now, then take care of removal after https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/669 ?

**Comment by rnveach** [comment]
Deprecating is fine, we should forbid usage by checkstyle now so we aren't adding new cases of it until it is removed. We can't remove it until we can cover all areas of CI that would be using it.

**Comment by strkkk** [comment]
I think this method can be removed if we do major version upgrade. There is no usages of this method in code
